src/main/scala/com/databricks/spark/sql/perf/mllib/BenchmarkAlgorithm.scala

```diff
@@ -67,9 +69,10 @@ trait ScoringWithEvaluator {
   final override def score(
       ctx: MLBenchContext,
       testSet: DataFrame,
-      model: Transformer): Double = {
+      model: Transformer): MLMetrics = {
     val eval = model.transform(testSet)
-    evaluator(ctx).evaluate(eval)
+    // TODO: add MetricName for evaluators
+    new MLMetrics("", evaluator(ctx).evaluate(eval), evaluator(ctx).isLargerBetter)
```

Comment:
Code reads better if we move `evaluator` part out. It triggers an action.

~~~scala
val results = model.transform(testSet)
val eval = evaluator(ctx)
val metricName = ...
val metricValue = eval.evaluate(eval)
MLMetric(metricName, metricValue, eval.isLargerBetter)
~~~

* minor: case class doesn't need `new`